src/pages/index.js

```diff
@@ -148,35 +184,80 @@ function getCardElement(data) {
   cardImage.setAttribute("src", `${data.link}`);
   cardImage.setAttribute("alt", `${data.name}`);
   cardTitle.textContent = `${data.name}`;
+  //to avoid loosing the like state after refresh
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
   cardLikeButton.addEventListener("click", (evt) => {
-    evt.preventDefault();
-    cardLikeButton.classList.toggle("card__like-button_liked");
+    // const isLiked = evt.target.classList.contains("card__like-button_liked");
+    api
+      .handleLike({ CardId: data._id, isLiked: data.isLiked })
+      .then((data) => {
+        evt.target.classList.toggle("card__like-button_liked");
+      })
+      .catch((err) => console.error(err));
   });
-  cardDeleteButton.addEventListener("click", () => {
-    // cardDeleteButton.closest(".card").remove();
-    card.remove();
+
+  cardDeleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(card, data._id);
   });
+
   cardImage.addEventListener("click", () => {
     modalPreviewCardImage.src = data.link;
     modalPreviewCardImage.alt = data.name;
     modalPreviewCaption.textContent = data.name;
     openModal(modalPreview);
   });
-
   return card;
 }
 
-// The function accepts a card object and a method of adding to the section
-// The method is initially `prepend`, but you can pass `append`
+function handleDeleteSubmit(evt) {
+  const button = evt.submitter;
+  setButtonText(button, true, "Deleting...", "Delete");
+  api;
+  api
+    .deleteCard({ id: selectedCardId })
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteCardModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      setButtonText(button, false, "Deleting", "Delete");
+    });
+}
+// Adding submit listener to the delete modal form on clicking delete button .
+deleteModalCardForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  handleDeleteSubmit(evt);
+});
+
 function renderCard(data, method = "prepend") {
   const cardElement = getCardElement(data);
   // Add the card into the section using the method
   cardsContainer[method](cardElement);
 }
 
-// a function that loops an array of objects and appends cards to our HTML.
-initialCards.forEach((data) => {
-  renderCard(data, (method = "append"));
+// Adding event listener to the profile avatar pencil-icon button .
+profileAvatarBtn.addEventListener("click", () => {
+  openModal(modalAvatar);
+});
+
+modalAvatarForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const button = evt.submitter;
+  setButtonText(button, true, "Saving...", "Save");
+  api
+    .editAvatarInfo({ avatar: modalAvatarInput.value })
+    .then((data) => {
+      avatar.setAttribute("src", `${data.avatar}`);
+      evt.target.reset();
```

Comment:
* Please, disable the `avatar` and `card` form submit buttons after submission in `.then`